tasks/tests/unit/test_preprocess_upload.py

```diff
@@ -216,6 +216,29 @@ def test_run_impl_unobtainable_lock(self, dbsession, mocker, mock_redis):
             "reason": "unable_to_acquire_lock",
         }
 
+    def test_get_repo_service_repo_and_owner_lack_bot(self, dbsession, mocker):
+        def mock_owner(foo):
+            print("in mock owner", foo)
+            raise OwnerWithoutValidBotError()
+
+        mock_owner_bot = mocker.patch(
+            "shared.bots.repo_bots.get_owner_or_appropriate_bot"
+        )
+        mock_owner_bot.side_effect = OwnerWithoutValidBotError()
```

Comment:
Technically this mock returns the *locally imported* `helpers.exceptions.OwnerWithoutValidBotError`, which could still exhibit the same problem (as in: being a different type from `shared.bots.exceptions.OwnerWithoutValidBotError`)

Reply:
this is a "gotcha" for mocking, but from what i was reading python still understands `OwnerWithoutValidBotError` is the same type regardless of the import path